src/Features/Core/IntroduceVariable/AbstractIntroduceVariableService.State.cs

```diff
@@ -145,6 +146,17 @@ public State(TService service, SemanticDocument document)
                     return false;
                 }
 
+                if (_service.IsInAutoPropertyInitializer(this.Expression))
```

Comment:
Good point. It works as expected, because block was preferred over initializer. I'll add a comment to make it clear and some tests as well to document the behavior.